src/pages/services/types/CustomerType.ts

```diff
@@ -0,0 +1,47 @@
+import {PageRequestType, PageResponseType} from "@/pages/services/types/PageType.ts";
+import {CertificateType} from "@/pages/services/types/CertificateType.ts";
+import {AuditType} from "@/pages/services/types/AuditType.ts";
+
+export interface CustomerType {
+    id: string;
+    name: string;
+    cpf: string;
+    brithDate: string;
+    phoneNumber: string;
+    email: string;
+    referenceNumber: string;
+    certificates: CertificateType[]
+    audit: AuditType
+}
+
+export type NewCustomerRequestType = Omit<CustomerType, 'id'>
+
+export interface SearchCustomerQueryParamsType extends PageRequestType {
+    id?: string;
+    name?: string;
+    cpf?: string;
+    brithDate?: string;
+    phoneNumber?: string;
+    email?: string;
+    referenceNumber?: string;
+    certificateId?: string;
+    certificateModel?: string;
+    certificateExpirationDate?: string;
+    certificateIssueDate?: string;
+}
+
+export interface CustomerWithNotificationType{
+    id?: string;
+    type?:string;
+    triggerDate?:string;
+    triggeredDate:string | null;
+    isManualTrigger?:boolean | null;
+    audit?:AuditType;
+}
```

Comment:
Esse type ta um pouquinho errado.   
Outro ponto, mesmo sendo errado, é bom se a tentar ao opcional, use o opcional somente em campos opcionais, na [wiki ](https://github.com/gina-system/gina-system-backend/wiki/Custumers-API#find-customer-and-notifications)tem uma tabelinha listando cada campo, nessa tabela fala exatamente quais campos são opcionais ou nao (coluna Required), se o campo do Response Body tiver Required SIM, ele sera retornado, logo nao é opcional.

A estrutura do retorno do Find Customer and Notifications é esse:
```json
{
    "data": {
        "customer": {
            "id": "e3e8de7c-d0b1-4d46-8c9a-df28e2e62526",
            "name": "Mateus Gomes da Silva Cardoso",
            "cpf": "85251056435",
            "brithDate": "1997-04-18T00:00:00.000Z",
            "phoneNumber": "+5561922222222",
            "email": "mateus@gmail.com",
            "referenceNumber": "39494944",
            "certificates": [
                {
                    "id": "1f7e4365-6931-47c2-8ee2-66260674fc3b",
                    "model": "A1",
                    "expirationDate": "2025-08-23T00:00:00.000Z",
                    "issueDate": "2024-08-23T00:00:00.000Z",
                    "audit": {
                        "createdAt": "2024-08-23T00:00:00.000Z",
                        "updatedAt": "2024-08-23T00:00:00.000Z"
                    }
                }
            ],
            "notifications": [
                {
                    "id": "79aae333-9045-4269-a2da-5d458b170b61",
                    "type": "BIRTHDAY",
                    "triggerDate": "2025-04-18T00:00:00.000Z",
                    "triggeredDate": null,
                    "isManualTrigger": null,
                    "audit": {
                        "createdAt": "2024-08-23T00:00:00.000Z",
                        "updatedAt": "2024-08-23T00:00:00.000Z"
                    }
                }
            ],
            "audit": {
                "createdAt": "2024-08-23T00:00:00.000Z",
                "updatedAt": "2024-08-23T00:00:00.000Z"
            }
        }
    }
}
```

Essa interface como está agora, gera o seguinte objeto:
```javascript
{ 
    customer: {
        audit: {
            createdAt: "",
            updatedAt: ""
        },
        triggerDate: "",
        isManualTrigger: true,
        type: "",
        triggeredDate: "",
        id: ""
    }
}
```
Da pra ver que está um pouco diferente do retornado da API.  

# Criando a interface de resposta
Se for criar uma interface que reflita o JSON retornado pelo endpoint, ficaria assim:   
Primeiro pode criar um arquivo `NotificationType.ts` na mesma pasta, nele vamos ter os types relacionados ao notification, dentro do arquivo criei esse interface que representa uma notificação:
```typescript
import {AuditType} from "@/pages/services/types/AuditType.ts";

export interface NotificationType {
    id: string;
    type: string;
    triggerDate: Date;
    triggeredDate?: Date
    isManualTrigger?: Date
    audit: AuditType
}
```

No arquivo `CustomerType.ts` atualize a interface CustomerWithNotificationType para ficar assim:
```typescript
import {NotificationType} from "@/pages/services/types/NotificationType.ts";

export interface CustomerWithNotificationType extends CustomerType {
    notifications: NotificationType[]
}
```
O retorno dessa API é um CustomerType, só que é um CustomerType com uma propriedade a mais, a lista de notifications. Já que ela é um CustomerType, a gente extends o CustomerType e adiciona a propriedade a mais